posydon/popsyn/synthetic_population.py

```diff
@@ -317,7 +320,7 @@ def parse(self, path_to_data, S1_state=None, S2_state=None, binary_state=None,
             sel_met = df_sel_met.index.drop_duplicates()
             
             if k > 0 and df_sel.shape[0] > 0:
-                shift_index = max(np.unique(df.index)) + 1 
+                shift_index += max(np.unique(df.index)) + 1 
```

Comment:
When introducing the `+` here, which makes sense, you should cover the case of `k > 0 and df_sel.shape[0] <= 0`, where we keep `shift_index` as is and not reset it to zero. I'd suggest this to become a non written else:
```
if k==0:
    shift_index = 0
elif df_sel.shape[0] > 0:
    shift_index += max(np.unique(df.index)) + 1
```
Usually, it won't matter, but we should may think, whether `k==0` or `k<=0` fits better our code style.

Reply:
Would it actually be better to always shift the index; independent if any binaries are selected?

That way we can define `shift_index` outside the metallicity loop and always increase for every metallicity in the Synthetic population.

Reply:
It depends on your needs:

- We need the shift index only to get unique IDs: no need to always shift
- We'd like to get reproducible and unique IDs: we should always shift

So, yes we can initialize the shift before the metallicity loop and increase it at the end of each metallicity.

Reply:
I have chosen to always shift for each metallicity. This is the best for reproducibility and debugging BinaryPopulations.